### ISSUE TITLE
[Config Support]: Issues after updating to 0.12

### ISSUE THREAD

**Issue by pott3r3r** [blakeblackshear/frigate#6286]
### Describe the problem you are having

Hello community!
So I've found myself in a hole, thats been dug over many years. I have somewhat a working config, but saw a youtube video of all the updates, so decided to update. Main issue here is prolly that my config from start was not perfect.

### Version

0.12.0-DA3E197

### Frigate config file

```yaml
ui:
  use_experimental: false
  live_mode: mse

mqtt:
  host: mqtt
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: homeassistant
  password: MySecretPassword
  stats_interval: 300

record:
  expire_interval: 10


go2rtc:
  streams:
    south:
      - rtsp://admin:MySecretPassword@192.168.50.23:554/h264Preview_01_main
    south_sub:
      - rtsp://admin:MySecretPassword@192.168.50.23:554/h264Preview_01_sub
    front:
      - rtsp://admin:MySecretPassword@192.168.50.24:554/h264Preview_01_main
    front_sub:
      - rtsp://admin:MySecretPassword@192.168.50.24:554/h264Preview_01_sub
    entrance:
      - rtsp://admin:MySecretPassword@192.168.50.25:554/h264Preview_01_main
    entrance_sub:
      - rtsp://admin:MySecretPasswordM@192.168.50.25:554/h264Preview_01_sub
    garage:
      - rtsp://admin:MySecretPassword@192.168.50.26:554/h264Preview_01_main
    garage_sub:
      - rtsp://admin:MySecretPassword@192.168.50.26:554/h264Preview_01_sub

timestamp_style:
  # Optional: Position of the timestamp (default: shown below)
  #           "tl" (top left), "tr" (top right), "bl" (bottom left), "br" (bottom right)
  position: "tl"
  # Optional: Format specifier conform to the Python package "datetime" (default: shown below)
  #           Additional Examples:
  #             german: "%d.%m.%Y %H:%M:%S"
  format: "%d/%m/%Y %H:%M:%S"
  # Optional: Color of font
  color:
    # All Required when color is specified (default: shown below)
    red: 255
    green: 255
    blue: 255
  # Optional: Line thickness of font (default: shown below)
  thickness: 2
  # Optional: Effect of lettering (default: shown below)
  #           None (No effect),
  #           "solid" (solid background in inverse color of font)
  #           "shadow" (shadow for font)
  effect: shadow

ffmpeg:
  hwaccel_args:
    - -c:v
    - h264_cuvid
  global_args: -hide_banner -loglevel warning
  # Optional: global input args (default: shown below)
  input_args: -c:v h264_cuvid -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -use_wallclock_as_timestamps 1
  # Optional: global output args
  output_args:
    detect: -f rawvideo -pix_fmt yuv420p 
    record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    rtmp: -c copy -f flv

cameras:
  front:
    ffmpeg:
      inputs:
        - path: rtsp://172.0.0.1:8554/front_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/front
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 640 
      height: 480
      fps: 5
    objects:
      track:
        - person
        - car
        - motorcycle
        - bicycle
        - cat
      filters:
        car:
          mask:
            - 111,0,640,0,640,90,640,145,605,134,552,121,476,107,410,92,361,81,292,69,235,65,159,93
            - 320,119,465,168,582,216,640,241,640,298,640,373,640,480,555,480,461,480,314,480,198,480,107,480,0,480,0,394,0,253,0,150,0,0,141,0,159,85
        cat:
          mask:
            - 460,0,414,0,377,0,344,0,289,0,203,0,0,0,0,156 
        person:
          mask:
            - 460,0,414,0,377,0,344,0,289,0,203,0,0,0,0,156
          threshold: 0.6
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    motion:
      mask:
        - 471,0,260,0,0,0,0,148
        - 327,43,406,55,481,69,553,93,596,103,640,120,640,152,564,127,467,104,381,84,265,66
        - 640,0,640,110,589,80,489,47,439,0
    record:
      enabled: true
      retain:
        days: 5
        mode: motion
      events:
        retain:
          default: 5
          mode: active_objects
  entrance:
    ffmpeg:
      inputs:
        - path: rtsp://172.0.0.1:8554/entrance_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/entrance
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 640
      height: 480
      fps: 5
    objects:
      track:
        - person
        - car
        - motorcycle
        - bicycle
        - cat
      filters:
        person:
          min_score: 0.4
          threshold: 0.5
          min_area: 700
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    motion:
      mask:
        - 47,53,79,47,115,71,209,86,317,98,386,136,480,183,365,480,640,480,640,286,640,203,640,129,640,0,482,0,320,0,141,0,0,0,0,60
    record:
      enabled: true
      retain:
        days: 5
        mode: motion
      events:
        required_zones:
          - zone_0
        retain:
          default: 5
          mode: active_objects
    zones:
      zone_0:
        coordinates: 231,75,156,42,61,35,0,67,0,119,52,151,98,168,144,250,351,160
      zone_1:
        coordinates: 640,281,640,480,573,480,370,480,163,283,375,166
  south:
    ffmpeg:
      inputs:
        - path: rtsp://172.0.0.1:8554/south_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/south
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 640
      height: 480
      fps: 5
    objects:
      track:
        - person
        - cat
      filters:
        person:
          min_score: 0.4
          threshold: 0.5
          min_area: 700
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    motion:
      mask:
        - 316,0,0,0,0,265
        - 565,0,270,0,272,25,561,47 
    record:
      enabled: true
      retain:
        days: 5
        mode: motion
      events:
        retain:
          default: 5
          mode: active_objects
  garage:
    ffmpeg:
      inputs:
        - path: rtsp://172.0.0.1:8554/garage_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/garage
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 640
      height: 480
      fps: 5
    objects:
      track:
        - person
        - motorcycle
        - bicycle
        - cat
      filters:
        person:
          min_score: 0.4
          threshold: 0.5
          min_area: 700
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    motion:
      mask:
        - 29,89,86,87,153,92,209,86,317,98,393,126,479,148,545,154,591,164,640,166,640,0,482,0,320,0,141,0,0,0,0,101
    record:
      enabled: true
      retain:
        days: 5
        mode: motion
      events:
        required_zones:
          - zone_0
        retain:
          default: 5
          mode: active_objects

detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu

rtmp:
  enabled: false

birdseye:
  enabled: True
  restream: False
  width: 1280 
  height: 720
  quality: 1
  mode: objects

live:
  height: 640
  quality: 1
```

### Relevant log output

```shell
*************************************************************
*************************************************************
***    Your config file is not valid!                     ***
***    Please check the docs at                           ***
***    https://docs.frigate.video/configuration/index     ***
*************************************************************
*************************************************************
***    Config Validation Errors                           ***
*************************************************************
22 validation errors for FrigateConfig
ffmpeg -> output_args -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> front -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> front -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> front -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> front -> height
  extra fields not permitted (type=value_error.extra)
cameras -> front -> width
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> entrance -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> height
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> width
  extra fields not permitted (type=value_error.extra)
cameras -> south -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> south -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> south -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> south -> height
  extra fields not permitted (type=value_error.extra)
cameras -> south -> width
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> woodpile -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> height
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> width
  extra fields not permitted (type=value_error.extra)
clips
  extra fields not permitted (type=value_error.extra)
Traceback (most recent call last):
  File "/opt/frigate/frigate/app.py", line 312, in start
    self.init_config()
  File "/opt/frigate/frigate/app.py", line 77, in init_config
    user_config = FrigateConfig.parse_file(config_file)
  File "/opt/frigate/frigate/config.py", line 903, in parse_file
    return cls.parse_obj(config)
  File "pydantic/main.py", line 511, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 331, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 22 validation errors for FrigateConfig
ffmpeg -> output_args -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> front -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> front -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> front -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> front -> height
  extra fields not permitted (type=value_error.extra)
cameras -> front -> width
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> entrance -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> height
  extra fields not permitted (type=value_error.extra)
cameras -> entrance -> width
  extra fields not permitted (type=value_error.extra)
cameras -> south -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> south -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> south -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> south -> height
  extra fields not permitted (type=value_error.extra)
cameras -> south -> width
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> ffmpeg -> inputs -> 1 -> roles -> 1
  value is not a valid enumeration member; permitted: 'record', 'rtmp', 'detect' (type=type_error.enum; enum_values=[<CameraRoleEnum.record: 'record'>, <CameraRoleEnum.rtmp: 'rtmp'>, <CameraRoleEnum.detect: 'detect'>])
cameras -> woodpile -> clips
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> fps
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> height
  extra fields not permitted (type=value_error.extra)
cameras -> woodpile -> width
  extra fields not permitted (type=value_error.extra)
clips
  extra fields not permitted (type=value_error.extra)

*************************************************************
***    End Config Validation Errors                       ***
*************************************************************
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

Other

### Any other information that may be helpful

nvidia-docker, gtx1060

**Comment by NickM-27** [comment]
your entire config besides the first line is indented incorrectly

**Comment by pott3r3r** [comment]
Not sure if it was a cut´n paste error, I dont see where it went wrong

**Comment by NickM-27** [comment]
it looks better now, it also seems like you were using tabs instead of spaces which is part of the problem.

Also I notice for some reason on all your sub streams you have `172.0.0.1` instead of `127.0.0.1`

**Comment by pott3r3r** [comment]
OMG, sharp! I will fix these and try again.

**Comment by pott3r3r** [comment]
It is indeed solved, only I changed was 172.0.0.1 instead of 127.0.0.1. Well I'm dumb, but at least there are ppl smart enough to spot my error :-)  
Thank you very much NickM-27!
